hack/test-prerequisites.go

```diff
@@ -55,5 +59,29 @@ func main() {
 		if err != nil {
 			log.Fatalf("Could not install %s CRD: %v", name, err)
 		}
+		err = wait.PollImmediate(time.Second, 30*time.Second, func() (bool, error) {
+			_, err = cvClient.ConfigV1().ClusterVersions().Create(ctx,
+				&configv1.ClusterVersion{
+					ObjectMeta: metav1.ObjectMeta{
+						Name: "version",
```

Comment:
The defaulting code we're removing had been using `Name: optr.name`, so these e2e-operator tests could drop in a ClusterVersion for testing without clobbering the ClusterVersion that the host cluster was using.  I think it works like this:

1. Host cluster comes up with `version` ClusterVersion.
2. `hack/test-integration.sh` called
3. New ClusterVersions created with different names that aren't `version`.
4. Some kind of integration testing on CVOs which consume those per-test ClusterVersions.
5. Testing completes.
6. Non-`version` ClusterVersions removed.

Or something.  I may be completely wrong :p

Reply:
moving conversation to [here][1].

[1]: https://github.com/openshift/cluster-version-operator/pull/741#discussion_r818372189